File: apps/admin/src/types/kyc.ts
export type KycDocumentType = "identity" | "address" | "source_of_funds" | "selfie";
export type KycDocumentStatus = "pending" | "verified" | "rejected" | "expired";
export type KycPriority = "low" | "medium" | "high";
export type KycReviewStatus = "pending" | "in_review" | "approved" | "rejected" | "resubmission_requested";
export type ScreeningStatus = "clear" | "pep_match" | "sanctions_hit" | "review_required";
export type SofStatus = "open" | "submitted" | "under_review" | "approved" | "rejected" | "expired";
export type RgAlertType = "chasing_losses" | "late_night_session" | "rapid_deposit_increase" | "long_session" | "limit_breach";

export interface KycDocument {
  id: string;
  player_id: string;
  type: KycDocumentType;
  status: KycDocumentStatus;
  file_url: string;
  uploaded_at: string;
  reviewed_by: string | null;
  reviewed_by_name: string | null;
  reviewed_at: string | null;
  rejection_reason: string | null;
  notes: string | null;
  expires_at: string | null;
  ocr_data: Record<string, unknown> | null;
}

export interface KycReviewItem {
  id: string;
  document_id: string;
  player_id: string;
  player_email: string;
  player_username: string;
  player_group: string;
  document_type: KycDocumentType;
  priority: KycPriority;
  status: KycReviewStatus;
  assigned_to: string | null;
  assigned_to_name: string | null;
  wait_time_minutes: number;
  reviewed_by: string | null;
  reviewed_at: string | null;
  decision_reason: string | null;
  created_at: string;
}

export interface KycQueueFilters {
  status?: KycReviewStatus;
  priority?: KycPriority;
  assigned_to?: string;
  document_type?: KycDocumentType;
  page?: number;
  page_size?: number;
}

export interface KycReviewPayload {
  decision: "approve" | "reject" | "resubmission";
  reason?: string;
  notes?: string;
}

export interface ScreeningResult {
  id: string;
  player_id: string;
  player_email: string;
  status: ScreeningStatus;
  matched_lists: string[];
  match_score: number;
  screened_at: string;
  next_screen_at: string | null;
  screened_by: string;
  reviewed_by: string | null;
  reviewed_at: string | null;
  review_notes: string | null;
}

export interface SofRequest {
  id: string;
  player_id: string;
  player_email: string;
  trigger_type: "threshold" | "single_tx" | "manual";
  threshold_amount: string | null;
  period_days: number | null;
  status: SofStatus;
  deadline_at: string;
  documents: SofDocument[];
  reviewed_by: string | null;
  reviewed_at: string | null;
  notes: string | null;
  created_at: string;
}

export interface SofDocument {
  type: "bank_statement" | "salary_slip" | "inheritance" | "property_sale" | "crypto_portfolio";
  file_url: string;
  uploaded_at: string;
}

export interface RgAlert {
  id: string;
  player_id: string;
  player_email: string;
  alert_type: RgAlertType;
  severity: "low" | "medium" | "high" | "critical";
  details: Record<string, unknown>;
  acknowledged_by: string | null;
  acknowledged_at: string | null;
  created_at: string;
}

export interface RgPlayerLimits {
  player_id: string;
  deposit_limit_daily: string | null;
  deposit_limit_weekly: string | null;
  deposit_limit_monthly: string | null;
  loss_limit: string | null;
  wager_limit_daily: string | null;
  session_time_limit_minutes: number | null;
  reality_check_frequency_minutes: number | null;
  self_exclusion_until: string | null;
  cool_off_until: string | null;
}

export interface ExpiryStats {
  expiring_30d: number;
  expiring_7d: number;
  expired: number;
}

export interface ExpiringDocument extends KycDocument {
  player_email: string;
  days_until_expiry: number;
}

export interface KycTeamMetric {
  officer_id: string;
  officer_name: string;
  metric_date: string;
  reviewed_count: number;
  avg_review_time_minutes: number;
  approve_count: number;
  reject_count: number;
  sla_breach_count: number;
}

export interface KycTeamStats {
  today: {
    queue_depth: number;
    avg_review_minutes: number;
    sla_breaches: number;
  };
  officers: KycTeamMetric[];
}

File: services/go/admin-bff/internal/models/kyc.go
package models

import (
	"encoding/json"
	"time"
)

type KycDocument struct {
	ID              string          `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID        int64           `gorm:"not null;index" json:"player_id"`
	Type            string          `gorm:"type:kyc_document_type;not null" json:"type"`
	Status          string          `gorm:"type:kyc_document_status;not null;default:'pending'" json:"status"`
	FileURL         string          `gorm:"type:varchar(512);not null" json:"file_url"`
	UploadedAt      time.Time       `gorm:"not null;default:now()" json:"uploaded_at"`
	ReviewedBy      *string         `gorm:"type:varchar(36)" json:"reviewed_by,omitempty"`
	ReviewedByName  *string         `gorm:"type:varchar(255)" json:"reviewed_by_name,omitempty"`
	ReviewedAt      *time.Time      `json:"reviewed_at,omitempty"`
	RejectionReason *string         `gorm:"type:varchar(500)" json:"rejection_reason,omitempty"`
	Notes           *string         `gorm:"type:text" json:"notes,omitempty"`
	ExpiresAt       *time.Time      `json:"expires_at,omitempty"`
	OcrData         json.RawMessage `gorm:"type:jsonb" json:"ocr_data,omitempty"`
	CreatedAt       time.Time       `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt       time.Time       `gorm:"not null;default:now()" json:"updated_at"`
}

type KycReview struct {
	ID               string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	DocumentID       string     `gorm:"type:uuid;not null;index" json:"document_id"`
	PlayerID         int64      `gorm:"not null;index" json:"player_id"`
	PlayerEmail      string     `gorm:"type:varchar(255);not null" json:"player_email"`
	PlayerUsername   string     `gorm:"type:varchar(100);not null" json:"player_username"`
	PlayerGroup      string     `gorm:"type:varchar(50);not null;default:'standard'" json:"player_group"`
	DocumentType     string     `gorm:"type:kyc_document_type;not null" json:"document_type"`
	Priority         string     `gorm:"type:kyc_priority;not null;default:'low'" json:"priority"`
	Status           string     `gorm:"type:kyc_review_status;not null;default:'pending'" json:"status"`
	AssignedTo       *string    `gorm:"type:varchar(36)" json:"assigned_to,omitempty"`
	AssignedToName   *string    `gorm:"type:varchar(255)" json:"assigned_to_name,omitempty"`
	WaitTimeMinutes  int        `gorm:"not null;default:0" json:"wait_time_minutes"`
	ReviewedBy       *string    `gorm:"type:varchar(36)" json:"reviewed_by,omitempty"`
	ReviewedAt       *time.Time `json:"reviewed_at,omitempty"`
	DecisionReason   *string    `gorm:"type:varchar(500)" json:"decision_reason,omitempty"`
	CreatedAt        time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt        time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type ScreeningResult struct {
	ID           string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID     int64      `gorm:"not null;index" json:"player_id"`
	PlayerEmail  string     `gorm:"type:varchar(255);not null" json:"player_email"`
	Status       string     `gorm:"type:screening_status;not null;default:'clear'" json:"status"`
	MatchedLists []string   `gorm:"type:text[]" json:"matched_lists"`
	MatchScore   int        `gorm:"not null;default:0" json:"match_score"`
	ScreenedAt   time.Time  `gorm:"not null;default:now()" json:"screened_at"`
	NextScreenAt *time.Time `json:"next_screen_at,omitempty"`
	ScreenedBy   string     `gorm:"type:varchar(36);not null;default:'system'" json:"screened_by"`
	ReviewedBy   *string    `gorm:"type:varchar(36)" json:"reviewed_by,omitempty"`
	ReviewedAt   *time.Time `json:"reviewed_at,omitempty"`
	ReviewNotes  *string    `gorm:"type:text" json:"review_notes,omitempty"`
	CreatedAt    time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt    time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type SofRequest struct {
	ID               string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID         int64      `gorm:"not null;index" json:"player_id"`
	PlayerEmail      string     `gorm:"type:varchar(255);not null" json:"player_email"`
	TriggerType      string     `gorm:"type:varchar(50);not null;default:'manual'" json:"trigger_type"`
	ThresholdAmount  *string    `gorm:"type:numeric(18,2)" json:"threshold_amount,omitempty"`
	PeriodDays       *int       `json:"period_days,omitempty"`
	Status           string     `gorm:"type:sof_status;not null;default:'open'" json:"status"`
	DeadlineAt       time.Time  `gorm:"not null" json:"deadline_at"`
	ReviewedBy       *string    `gorm:"type:varchar(36)" json:"reviewed_by,omitempty"`
	ReviewedAt       *time.Time `json:"reviewed_at,omitempty"`
	Notes            *string    `gorm:"type:text" json:"notes,omitempty"`
	CreatedAt        time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt        time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type SofDocument struct {
	ID         string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	RequestID  string    `gorm:"type:uuid;not null;index" json:"request_id"`
	Type       string    `gorm:"type:sof_document_type;not null" json:"type"`
	FileURL    string    `gorm:"type:varchar(512);not null" json:"file_url"`
	UploadedAt time.Time `gorm:"not null;default:now()" json:"uploaded_at"`
}

type RgAlert struct {
	ID              string          `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID        int64           `gorm:"not null;index" json:"player_id"`
	PlayerEmail     string          `gorm:"type:varchar(255);not null" json:"player_email"`
	AlertType       string          `gorm:"type:rg_alert_type;not null" json:"alert_type"`
	Severity        string          `gorm:"type:rg_alert_severity;not null;default:'medium'" json:"severity"`
	Details         json.RawMessage `gorm:"type:jsonb;not null;default:'{}'" json:"details"`
	AcknowledgedBy  *string         `gorm:"type:varchar(36)" json:"acknowledged_by,omitempty"`
	AcknowledgedAt  *time.Time      `json:"acknowledged_at,omitempty"`
	CreatedAt       time.Time       `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt       time.Time       `gorm:"not null;default:now()" json:"updated_at"`
}

type RgLimit struct {
	ID                           string     `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	PlayerID                     int64      `gorm:"not null;uniqueIndex" json:"player_id"`
	DepositLimitDaily            *string    `gorm:"type:numeric(18,2)" json:"deposit_limit_daily,omitempty"`
	DepositLimitWeekly           *string    `gorm:"type:numeric(18,2)" json:"deposit_limit_weekly,omitempty"`
	DepositLimitMonthly          *string    `gorm:"type:numeric(18,2)" json:"deposit_limit_monthly,omitempty"`
	LossLimit                    *string    `gorm:"type:numeric(18,2)" json:"loss_limit,omitempty"`
	WagerLimitDaily              *string    `gorm:"type:numeric(18,2)" json:"wager_limit_daily,omitempty"`
	SessionTimeLimitMinutes      *int       `json:"session_time_limit_minutes,omitempty"`
	RealityCheckFrequencyMinutes *int       `json:"reality_check_frequency_minutes,omitempty"`
	SelfExclusionUntil         *time.Time `json:"self_exclusion_until,omitempty"`
	CoolOffUntil               *time.Time `json:"cool_off_until,omitempty"`
	CreatedAt                  time.Time  `gorm:"not null;default:now()" json:"created_at"`
	UpdatedAt                  time.Time  `gorm:"not null;default:now()" json:"updated_at"`
}

type KycTeamMetric struct {
	ID                    string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	OfficerID             string    `gorm:"type:varchar(36);not null" json:"officer_id"`
	OfficerName           string    `gorm:"type:varchar(255);not null" json:"officer_name"`
	MetricDate            time.Time `gorm:"type:date;not null" json:"metric_date"`
	ReviewedCount         int       `gorm:"not null;default:0" json:"reviewed_count"`
	AvgReviewTimeMinutes  int       `gorm:"not null;default:0" json:"avg_review_time_minutes"`
	ApproveCount          int       `gorm:"not null;default:0" json:"approve_count"`
	RejectCount           int       `gorm:"not null;default:0" json:"reject_count"`
	SlaBreachCount        int       `gorm:"not null;default:0" json:"sla_breach_count"`
}
